Remove shared `getDockerBuildTask` to enable parallel docker builds (#16384)
airbytehq/airbyte: docs/contributing-to-airbyte/developing-on-docker.md

The docker build is fully incremental for the platform build, which means that it will only build an image if it is needed. We need to keep it that 
way.

> The top level `build.gradle` file defines several convenient tasks for building a docker image.

Reviewer: @git-phu how does this look?

Reviewer 2: looks great! Just one comment

Reviewer: Thanks!

1) The `copyGeneratedTar` task copies a generated TAR file from a default location into the default location used by the [docker plugin](https://github.com/bmuschko/gradle-docker-plugin).
2) The `buildDockerImage` task is a convenience class for configuring the above linked docker plugin that centralizes configuration logic commonly found in our dockerfiles.
3) Makes the `buildDockerImage` task depend on the Gradle `assemble` task.

> These tasks are added to a subproject if the subproject has a `gradle.properties` file with the `dockerImageName` property. This property is sets the build docker image's name.

Reviewer 2: ```suggestion
These tasks are created in a subproject if the subproject has a `gradle.properties` file with the `dockerImageName` property. This property sets the built docker image's name.
```
small typos
not sure about saying created vs added, but I wanted it to be clear that each subproject gets its own `copyGeneratedTar` and `buildDockerImage` task now instead of how they used to be root level tasks (and with multiple versions in the case of `buildDockerImage-blah`
